packages/e2e-tests/specs/editor/various/preview.test.js

```diff
@@ -145,9 +144,9 @@ describe( 'Preview', () => {
 		expect( previewTitle ).toBe( 'Hello World! And more.' );
 
 		// Published preview URL should include ID and nonce parameters.
-		const { query } = parse( previewPage.url(), true );
-		expect( query ).toHaveProperty( 'preview_id' );
-		expect( query ).toHaveProperty( 'preview_nonce' );
+		const { searchParams } = new URL( previewPage.url() );
```

Comment:
Once again, this will only work for absolute URLs, and will throw an exception for everything else. If you want to support non-absolute URLs, you should use a base URL. In this case, you wouldn't even have to look at whether the hostname or protocol were taken from the base URL, since all you're interested in is the search params:

```js
const { searchParams } = new URL( previewPage.url(), 'https://example.com' );
```

The `try`/`catch` may still be needed if you want to handle invalid URLs without letting an exception bubble up.

Reply:
`previewPage.url()` is the result of [`page.url`](https://github.com/puppeteer/puppeteer/blob/v2.0.0/docs/api.md#pageurl) ([`frame.url`](https://github.com/puppeteer/puppeteer/blob/v2.0.0/docs/api.md#frameurl)). I'm not positive, but I think it should be safe to assume it will be an absolute URL as reported by the browser.

Reply:
> `previewPage.url()` is the result of [`page.url`](https://github.com/puppeteer/puppeteer/blob/v2.0.0/docs/api.md#pageurl) ([`frame.url`](https://github.com/puppeteer/puppeteer/blob/v2.0.0/docs/api.md#frameurl)). I'm not positive, but I think it should be safe to assume it will be an absolute URL as reported by the browser.

Yes, that's probably a safe assumption 👍